### DISCUSSION TITLE
Belkin charger not negotiating USB PD

### DISCUSSION THREAD

**Discussion by TomW1605** [Ralim/IronOS#1890] [General]
continuation of a conversation in discord (see screenshot)

i have a Belkin 45W F7U010au that seems to only be giving 5V and the pinecil v2 is rebooting every few seconds. in the PD debug screen it cycles 3>6>11>reboot>repeat.

the origonal page on the Belkin site is gone but reviews say it is a PD charger and it has the right voltages (5, 9, 12 and 15, although im not sure 12V is normal).

![image](https://github.com/Ralim/IronOS/assets/17092573/d680e5d0-d8c4-47bf-99dc-64cdbd581c06)

**Comment by vulpes2** [comment]
Belkin chargers seem pretty buggy in general, I have a different charger that is causing problems as well. On the WCH001, the Pinecil v1 will reliably negotiate 15V, but the Pinecil v2 will only get 9V most of the time, despite having the full list of power configurations listed in the PD debug menu. Both devices are on v2.22.

The WCH001 supports the following power configurations (verified with a USB PD meter):
- 5V 3A
- 9V 3A
- 15V 2A
- 20V 1.5A

https://www.belkin.com/p/30w-usb-c-pd-gan-wall-charger/P-WCH001.html

**Reply by Ralim** [reply]
@vulpes2 what your seeing isn't a bug.

The V2 tips are 6.2 ohm not 8 so to be able to use a voltage it will require a higher current than the older 8 ohm tips. This is the correct behaviour to avoid overloading the power supply.

**Reply by vulpes2** [reply]
So if the Pinecil v2 selects 15V instead of 9V on this particular charger, it's supposed to be a bug?

**Reply by Ralim** [reply]
So if your using this supply, depending on the tip type installed when it's powered up and connected it should select the correct voltage.

Long tips are 8 ohms, pine short tips are 6.2.

Using ohms law. V=I*R, we can rearrange to Voltage/Resistance = current rating required.

For a short tip (6.2 ohm)
15V requires 2.4A
9V requires 1.45A

So it should pick 9V for this charger

Whereas a long tip (8 ohms)
15V requires 1.8A
So in this case, it would select 15V as it's the highest voltage that can also deliver the required current.


The short tips (6.2ohm) are really optimised for 45/60/90/100/240W chargers.
